radis/lbl/base.py

```diff
@@ -2400,22 +2686,46 @@ def _calc_populations_noneq_multiTvib(
         # ... Total
         if rot_distribution == "boltzmann":
             # ... total
-            df["nu"] = (
-                nu_vib1Qvib1
-                * nu_vib2Qvib2
-                * nu_vib3Qvib3
-                * df.grotu
-                * exp(-df.Erotu * hc_k / Trot)
-                / df.Q
-            )
-            df["nl"] = (
-                nl_vib1Qvib1
-                * nl_vib2Qvib2
-                * nl_vib3Qvib3
-                * df.grotl
-                * exp(-df.Erotl * hc_k / Trot)
-                / df.Q
-            )
+
+            if "iso" in df:
+
+                df["nu"] = (
+                    nu_vib1Qvib1
+                    * nu_vib2Qvib2
+                    * nu_vib3Qvib3
+                    * df.grotu
+                    * exp(-df.Erotu * hc_k / Trot)
+                    / df["iso"].map(Q_dict)
+                )
+                df["nl"] = (
+                    nl_vib1Qvib1
+                    * nl_vib2Qvib2
+                    * nl_vib3Qvib3
+                    * df.grotl
+                    * exp(-df.Erotl * hc_k / Trot)
+                    / df["iso"].map(Q_dict)
+                )
+
+            else:
+
+                Q_dict = list(Q_dict.values())
+
+                df["nu"] = (
+                    nu_vib1Qvib1
+                    * nu_vib2Qvib2
+                    * nu_vib3Qvib3
+                    * df.grotu
+                    * exp(-df.Erotu * hc_k / Trot)
+                    / Q_dict[0]
+                )
+                df["nl"] = (
+                    nl_vib1Qvib1
+                    * nl_vib2Qvib2
+                    * nl_vib3Qvib3
+                    * df.grotl
+                    * exp(-df.Erotl * hc_k / Trot)
+                    / Q_dict[0]
+                )
```

Comment:
So, I though it would be difficult to generalise the if/else blocks for both eq and non_eq so I have split the handling of Q to two functions. One of those functions is identical to the psuedo code as you suggested. I have used this function in the non_eq handling as well. If you approve of these changes, I will add docstrings and comments.

Reply:
It looks very good, and having one for eq and one for noneq makes a lot of sense.